stable_baselines/gail/dataset/dataset.py

```diff
@@ -109,62 +180,222 @@ def __init__(self, expert_path=None, traj_data=None, train_fraction=0.7, batch_s
 
-    def __del__(self):
-        del self.dataloader, self.train_loader, self.val_loader
 
-    def prepare_pickling(self):
-        """
-        Exit processes in order to pickle the dataset.
-        """
-        self.dataloader, self.train_loader, self.val_loader = None, None, None
 
-    def log_info(self):
-        """
-        Log the information of the dataset.
-        """
-        logger.log("Total trajectories: {}".format(self.num_traj))
-        logger.log("Total transitions: {}".format(self.num_transition))
-        logger.log("Average returns: {}".format(self.avg_ret))
-        logger.log("Std for returns: {}".format(self.std_ret))
+class ExpertDatasetLSTM(Dataset):
+    """
+    Dataset for using behavior cloning or GAIL.
 
-    def get_next_batch(self, split=None):
-        """
-        Get the batch from the dataset.
+    The structure of the expert dataset is a dict, saved as an ".npz" archive.
+    The dictionary contains the keys 'actions', 'episode_returns', 'rewards',
+    'obs' and 'episode_starts'. The corresponding values have data
+    concatenated across episode: the first axis is the timestep,
+    the remaining axes index into the data. In case of images,
+    'obs' contains the relative path to the images, to enable space
+    saving from image compression.
+
+    :param expert_path: (str) The path to trajectory data (.npz file).
+        Mutually exclusive with traj_data.
+    :param traj_data: (dict) Trajectory data, in format described above.
+        Mutually exclusive with expert_path.
+    :param train_fraction: (float) the train validation split (0 to 1)
+        for pre-training using behavior cloning (BC)
+    :param batch_size: (int) the minibatch size for behavior cloning
+    :param traj_limitation: (int) the number of trajectory to use (if -1, load all)
+    :param verbose: (int) Verbosity
+    :param sequential_preprocessing: (bool) Do not use subprocess to preprocess
+        the data (slower but use less memory for the CI)
+    :param envs_per_batch: (int) Only used if LSTM is True. Number of envs that
+        are processed per batch.
+    """
 
-        :param split: (str) the type of data split (can be None, 'train', 'val')
-        :return: (np.ndarray, np.ndarray) inputs and labels
-        """
-        dataloader = {
-            None: self.dataloader,
-            'train': self.train_loader,
-            'val': self.val_loader
-        }[split]
+    def __init__(self, expert_path=None, traj_data=None, train_fraction=0.7,
+                 batch_size=64, traj_limitation=-1, verbose=1, envs_per_batch=1,
+                 sequential_preprocessing=False):
 
-        if dataloader.process is None:
-            dataloader.start_process()
-        try:
-            return next(dataloader)
-        except StopIteration:
-            dataloader = iter(dataloader)
-            return next(dataloader)
+        if traj_data is not None and expert_path is not None:
```

Comment:
I originally had it in one class, but someone who used the PR, has suggested to split it in two classes. I think that this was a good idea, because it clearly improved the user friendliness of the `ExpertDataset` class.